pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1577,13 +1577,23 @@ func (og *operationGenerator) GenerateExpandVolumeFunc(
 func (og *operationGenerator) GenerateExpandVolumeFSWithoutUnmountingFunc(
 	volumeToMount VolumeToMount,
 	actualStateOfWorld ActualStateOfWorldMounterUpdater) (volumetypes.GeneratedOperations, error) {
-	volumePlugin, err :=
-		og.volumePluginMgr.FindPluginBySpec(volumeToMount.VolumeSpec)
-	if err != nil || volumePlugin == nil {
-		return volumetypes.GeneratedOperations{}, volumeToMount.GenerateErrorDetailed("VolumeFSResize.FindPluginBySpec failed", err)
-	}
 
 	fsResizeFunc := func() (error, error) {
+		// Need to translate the spec here if the plugin is migrated so that the metrics
+		// emitted show the correct (migrated) plugin
+		if useCSIPlugin(og.volumePluginMgr, volumeToMount.VolumeSpec) {
+			csiSpec, err := translateSpec(volumeToMount.VolumeSpec)
+			if err != nil {
+				return volumeToMount.GenerateError("VolumeFSResize.translateSpec failed", err)
+			}
+			volumeToMount.VolumeSpec = csiSpec
+		}
+		volumePlugin, err :=
+			og.volumePluginMgr.FindPluginBySpec(volumeToMount.VolumeSpec)
+		if err != nil || volumePlugin == nil {
+			return volumeToMount.GenerateError("VolumeFSResize.FindPluginBySpec failed", err)
+		}
```

Comment:
If we need provide detailed error info when volumePlugin is nil?

Reply:
You mean like code below? I think `GenerateError` works as well here. Also it automatically returns two error values and hence is easier to satisfy constraints of the function.